frontend/packages/console-shared/src/hooks/useUserSettings.ts

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+// FIXME upgrading redux types is causing many errors at this time
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useSelector } from 'react-redux';
+import { RootState } from '@console/internal/redux';
+import { ConfigMapModel, ProjectRequestModel, ProjectModel } from '@console/internal/models';
+import { K8sResourceKind, k8sGet, k8sCreate, k8sPatch } from '@console/internal/module/k8s';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+
+// can't create project with name prefix with 'openshift-*', once we have proxy need to update
+const USER_SETTING_CONFIGMAP_NAMESPACE = 'console-user-settings';
+
+const getConfigResourceData = (userUid: string, key: string, defaultValue: string) => {
+  return [
+    {
+      kind: ConfigMapModel.kind,
+      namespace: USER_SETTING_CONFIGMAP_NAMESPACE,
+      isList: false,
+      name: `user-settings-${userUid}`,
+    },
+    {
+      apiVersion: ConfigMapModel.apiVersion,
+      kind: ConfigMapModel.kind,
+      metadata: {
+        name: `user-settings-${userUid}`,
+        namespace: USER_SETTING_CONFIGMAP_NAMESPACE,
+      },
+      data: {
+        [key]: defaultValue,
+      },
+    },
+  ];
+};
+
+// This won't be needed once we have proxy api
+const getProject = async () => {
+  try {
+    await k8sGet(ProjectModel, USER_SETTING_CONFIGMAP_NAMESPACE);
+  } catch {
+    await k8sCreate(ProjectRequestModel, {
+      metadata: {
+        name: USER_SETTING_CONFIGMAP_NAMESPACE,
+      },
+    });
+  }
+};
+
+const createConfigMap = async (configMapData: K8sResourceKind) => {
+  try {
+    await k8sCreate(ConfigMapModel, configMapData);
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error(err);
+  }
+};
+
+const patchConfigMap = async (patch, cfData) => {
```

Comment:
```suggestion
const updateConfigMap = async (configMap: ConfigMapKind, key: string, value: string) => {
 // create the patch here inline
```

This would simplify the useEffect which triggers an update